### PR TITLE
Add key import flag to specify if new pass needed

### LINKED ISSUE #4369: Newpair: should mimic 'gpg --import'
## Version of Singularity:

```
v3.4
```

### Expected behavior

When importing a private key, by default it should not ask to set a new password. That should be a separate flag like `--with-newpassword`.

### Actual behavior

As of now, `import` will ask to set a newpassword when importing a key.

### Steps to reproduce behavior

Write here.

### PR DESCRIPTION
**Pull request by pisarukv**
closes https://github.com/sylabs/singularity/issues/4369

### CODE REVIEW

**Review by sashayakovtseva — APPROVED** [review]

File: cmd/internal/cli/key_import.go [hunk 1]
```diff
@@ -8,6 +8,8 @@ package cli
 import (
 	"os"
 
+	"github.com/sylabs/singularity/pkg/cmdline"
+
```
Comment (WestleyK) [inline cmd/internal/cli/key_import.go]: Extra newline

```suggestion
```

File: cmd/internal/cli/key_import.go [hunk 2]
```diff
@@ -27,9 +29,18 @@ var KeyImportCmd = &cobra.Command{
 	Example: docs.KeyImportExample,
 }
 
+var keyImportWithNewPassword bool
+var keyImportWithNewPasswordFlag = cmdline.Flag{
+	ID:           "keyImportWithNewPasswordFlag",
+	Value:        &keyImportWithNewPassword,
+	DefaultValue: false,
+	Name:         "with-newpassword",
```
Comment (WestleyK) [inline cmd/internal/cli/key_import.go]: Maybe just `new-password` will be better?

```suggestion
	Name:         "new-password",
```

File: cmd/internal/cli/key_import.go [hunk 3]
```diff
@@ -27,9 +29,18 @@ var KeyImportCmd = &cobra.Command{
...
+	Usage:        `if specified will ask to set a new password`,
```
Comment (WestleyK) [inline cmd/internal/cli/key_import.go]: ~Maybe?...~

```
	Usage:        `if specified, newpair will ask to set a new password`,
```

**Review by WestleyK — CHANGES_REQUESTED** [review]
Have a couple suggestions, otherwise looks good.

File: cmd/internal/cli/key_import.go (hunk 3 above)
Comment (WestleyK) [reply cmd/internal/cli/key_import.go]: Or better:

```suggestion
	Usage:        `set a new password to the private key`,
```

**Review by WestleyK — COMMENTED** [review]
We will also need to update the E2E tests to not send to many passwords.

File: pkg/sypgp/sypgp.go [hunk 4]
```diff
@@ -1014,38 +1018,35 @@ func (keyring *Handle) importPrivateKey(entity *openpgp.Entity) error {
...
-
-	// Make a clone of the entity
-	newEntity := &openpgp.Entity{
-		PrimaryKey:  entity.PrimaryKey,
-		PrivateKey:  entity.PrivateKey,
-		Identities:  entity.Identities,
-		Revocations: entity.Revocations,
-		Subkeys:     entity.Subkeys,
-	}
+	newEntity := *entity
```
Comment (WestleyK) [inline pkg/sypgp/sypgp.go:1021]: Why make a `newEntity`? can you just use `entity`?

File: pkg/sypgp/sypgp.go [hunk 5]
```diff
@@ -1014,38 +1018,35 @@ func (keyring *Handle) importPrivateKey(entity *openpgp.Entity) error {
...
-		return err
+	if password != "" {
+		if err := newEntity.PrivateKey.Encrypt([]byte(password)); err != nil {
+			return err
+		}
 	}
 
 	// Store the private key
-	if err := keyring.appendPrivateKey(newEntity); err != nil {
+	if err := keyring.appendPrivateKey(&newEntity); err != nil {
```
Comment (WestleyK) [inline pkg/sypgp/sypgp.go:1049]: Why is the `&` here? Is it also necessary for the other calls to `keyring.appendPubKey`, and `keyring.appendPrivateKey`?

**Review by WestleyK — CHANGES_REQUESTED** [review]
This behaves much nicer! Just a couple comment/suggestions. Also, the e2e tests need to be updated (the `import` tests only need to send the password once)

File: pkg/sypgp/sypgp.go (hunk 4 above)
Comment (pisarukv) [reply pkg/sypgp/sypgp.go:1021]: So for some reasons here was used a copy of entity. I decided to leave it as is. I'll investigate deeper if we can use entity directly.

File: pkg/sypgp/sypgp.go (hunk 5 above)
Comment (sashayakovtseva) [reply pkg/sypgp/sypgp.go:1049]: @WestleyK `&` is a pointer gian operation. It is there b/c `keyring.appendPubKey` accepts pointer, not value.

**Review by WestleyK — APPROVED** [review]
This looks good! We can update the E2E tests later. _Also maybe worth squashing the commits?..._

**Pull request merged**